tests/unit_tests/epee_serialization.cpp

```diff
@@ -77,6 +77,67 @@ struct ObjOfInts
     KV_SERIALIZE(x)
   END_KV_SERIALIZE_MAP()
 };
+
+template<typename t_param>
+struct ParentObjWithOptChild
+{
+  t_param     params;
+
+  ParentObjWithOptChild(): params{} {}
+
+  BEGIN_KV_SERIALIZE_MAP()
+    KV_SERIALIZE(params)
+  END_KV_SERIALIZE_MAP()
+};
+
+struct ObjWithOptChild
+{
+  bool test_value;
+
+  BEGIN_KV_SERIALIZE_MAP()
+    KV_SERIALIZE_OPT(test_value, true);
+  END_KV_SERIALIZE_MAP()
+};
+}
+
+TEST(epee_binary, serialize_deserialize)
+{
+  ParentObjWithOptChild<ObjWithOptChild> o;
+  std::string o_json;
+  o.params.test_value = true;
+
+  EXPECT_TRUE(epee::serialization::store_t_to_json(o, o_json));
+  EXPECT_TRUE(o.params.test_value);
+
+  EXPECT_TRUE(epee::serialization::load_t_from_json(o, o_json));
+  EXPECT_TRUE(o.params.test_value);
+
+  ParentObjWithOptChild<ObjWithOptChild> o2;
+  std::string o2_json;
+  o.params.test_value = false;
+
+  EXPECT_TRUE(epee::serialization::store_t_to_json(o2, o2_json));
+  EXPECT_FALSE(o2.params.test_value);
+
+  EXPECT_TRUE(epee::serialization::load_t_from_json(o2, o2_json));
+  EXPECT_FALSE(o2.params.test_value);
+
+  // compiler sets default value of test_value to false
+  ParentObjWithOptChild<ObjWithOptChild> o3;
+  std::string o3_json;
+
+  EXPECT_TRUE(epee::serialization::store_t_to_json(o3, o3_json));
+  EXPECT_FALSE(o3.params.test_value);
+
+  EXPECT_TRUE(epee::serialization::load_t_from_json(o3, o3_json));
+  EXPECT_FALSE(o3.params.test_value);
+
+  // test case with empty json, to test default value initialization
+  ParentObjWithOptChild<ObjWithOptChild> o4;
+  std::string o4_json = "{}";
+
+  EXPECT_TRUE(epee::serialization::load_t_from_json(o4, o4_json));
+  EXPECT_TRUE(o4.params.test_value);
```

Comment:
This is a bad test, this test requires that the `params` fields in `ParentObjWithOptChild` is optional (it is not specified as optional).

This test should make `load_t_from_json` error as the `params` field is missing and it is not an optional field.

Reply:
I see what you saying, but the point is we don't have ( or require ) param in like substantial of portion of our RPC commands. 

But if we are not able to initialize the `params` it can have serious consequences. Like the one I explained for `close_wallet`.

1. https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#close_wallet

Reply:
`params` here is not the same as the `params` in the JSON-RPC it is just the name of the field in the test struct.

`params` in the real JSON-RPC code should be an optional field, like what #9573 does.

Reply:
Yes, you are right. I am closing this PR. and re-opening our #9573 PR.